src/construction.cpp

```diff
@@ -472,8 +485,13 @@ void construction_menu()
 
         const std::string action = ctxt.handle_input();
         const long raw_input_char = ctxt.get_raw_input().get_first_input();
-
-        if( action == "DOWN" ) {
+        if( action == "FILTER" ){
+            update_info = true;
+            update_cat = true;
+            tabindex = 9;
```

Comment:
Should probably not switch until player inputs a filter. That is, cancel or empty filter should avoid switching.

Could be better if filter input filtered things in real time, though it also could be worse if it caused lag.